super_dragon_magical_unicorns_by_disney_fish/server.js

```diff
@@ -0,0 +1,86 @@
+var express = require('express');
+var bodyParser = require('body-parser');
+var path = require('path');
+var app = express();
+var mongoose = require('mongoose');
+
+// connect this to the db
+mongoose.connect('mongodb://localhost/dragon_dashboard')
+
+//create Schema
+
+var DragonSchema = new mongoose.Schema({
+  name: String,
+  age: Number,
+  color: String
+})
+
+mongoose.model('Dragon', DragonSchema);
+var Dragon = mongoose.model('Dragon')
+
+app.use(bodyParser.urlencoded({extended:true}));
+app.use(express.static(path.join(__dirname, './static')));
+app.set('views', path.join(__dirname, './views'));
+app.set('view engine', 'ejs');
+
+// Routes
+app.get('/', function(req, res){
+  Dragon.find({}, function(err, dragons){
+    console.log(dragons);
+    if (err){
+      console.log("something broke");
+    }
+    else{
+      console.log("yay dragons");
+      res.render('index', {"dragons": dragons});
+    }
+  })
+})
+app.post('/', function(req, res){
+  var dragon = new Dragon ({name: req.body.name, age: req.body.age, color: req.body.color});
+  dragon.save(function(err){
+    if(err){
+      console.log("couldn't add Dragon");
+    }else{
+      console.log("Dragon Added");
+      res.redirect('/');
+    }
+  })
+})
+
+app.get('/new', function(req, res){
+  res.render('new')
+})
+
+app.get('/:id/edit', function(req, res){
+  Dragon.find({ _id: req.params.id }, function(err, dragon){
```

Comment:
Using findOne will prevent you from having to fill your view with instances like `dragon[0]`. Alternatively, we can just send `dragon[0]` in the render method, and you won't need to repeat it a bunch of times on your view!